### ISSUE TITLE
How to remove particular file patterns from code coverage files

### ISSUE THREAD

**Issue by Yahsaswi** [SlatherOrg/slather#364]
I would like to remove .h files from being included in code coverage. I tried with the regex "--ignore ../**/.h ", but that doesn't work. Could someone help me with this.

**Comment by MacGritsch** [comment]
Why? If it contains executable code, you should have a test for it.
In most cases you can also move it to the implementation file.

**Comment by Yahsaswi** [comment]
Yeah. True, but the Xcode coverage report itself doesn't include .h files. So, thought that would be a right practice. However, there are some particular files that I wish to exclude from coverage files. So would like to know if there a procedure how we can do the same.

**Comment by MacGritsch** [comment]
It does - if they contain code.

There is an ignore paramter for slather. so you can specify that files there (also headers if you want).

**Comment by Yahsaswi** [comment]
Yeah. So "--ignore ../**/.h" didn't work. Wanted to understand what's the exact syntax

**Comment by micnguyen** [comment]
This is a use case I am also after (not specifically .h files, but having a regexp matcher on excluding files). My use case is that due to the architecture of our app, `*ViewController.swift`files are not useful to unit test so we would like our coverage to reflect that.

**Comment by vitorll** [comment]
I have the exactly same need as @micnguyen. Any ideas?

**Comment by jeffsorr** [comment]
With slather 2.4.6 the following works for me:
`slather coverage -s --ignore "*.h" --scheme myScheme ./myProject.xcodeproj`

**Comment by jeffsorr** [comment]
Also note that you can use multiple --ignore args e.g.
`slather coverage -s --ignore "*.h" --ignore "thirdparty/*" --scheme myScheme ./myProject.xcodeproj`.  It seems --ignore only honors the * wildcard, not true regex or *nix path operators, so this will **not** work: `slather coverage -s --ignore "./thirdparty/*" --scheme myScheme ./myProject.xcodeproj`

**Comment by uhooi** [comment]
If you use a wildcard at the beginning, it must be enclosed in double quotes( `"` ).

e.g.
.slather.yml

```yaml
ignore:
  - Pods/*
  - Carthage/*
  - "**/AppDelegate.swift"
  - "**/SceneDelegate.swift"
  - "**/Generated/*"
  - "**/UIParts/*"
  - "**/Views/*"
  - "**/Routers/*"
```

**Comment by uhooi** [comment]
I want to ignore the `SourcePackages` folder and below, but I can't 😢 
ref: #475
